pattoo/api/portal/routes/datapoints.py

```diff
@@ -0,0 +1,89 @@
+"""Routes for datapoints."""
+
+# Flask imports
+from flask import Blueprint
+from flask import request, session
+from sqlalchemy import and_
+
+# Pattoo imports
+from pattoo.db import db
+from pattoo.db.models import DataPoint as DataPModel
+from pattoo.db.models import PairXlate as PairModel
+
+# Define the AGENTS global variable
+DATAPOINTS = Blueprint('DATAPOINTS', __name__)
+
+@DATAPOINTS.route('/api/datapoints', methods=['GET'])
+def datapoints():
+    """
+    Manage datapoints for an agent.
+
+    Args:
+        None
+    
+    Returns:
+        response (dict): Response Message
+
+    """
+
+    pp_datapoints = [] # Store formatted list
+
+    # Check if a user was stored in session first
+    if session.get('idx_user', None) is None:
+        response = {'data': {'message': 'Login first'}}
+        # Block access if no user was found in session
+        return response
+
+    enabled = request.args.get('enabled', type=int)
+    idx_agent = request.args.get('idx_agent', type=int)
+    idx_datapoint = request.args.get('idx_datapoint', type=int)
+    response = {'data':{'message': 'Query did not run'}}
```

Comment:
Initialize **response** this at the top of your function for reasons stated previously. Always do this. It's a bad practice otherwise.